### PR TITLE
Dashboard tables

### PR DESCRIPTION
**Pull request by patrickwhardy**
Add upcoming and past reservation tables to user dashboard and methods in user model
add homes' upcoming and past tables and methods in homes model
add "Add a Home" button to dashboard

closes issue #7 
closes issue #16

### CODE REVIEW

File: app/models/home.rb [hunk 1]
```diff
@@ -4,6 +4,15 @@ class Home < ActiveRecord::Base
...
   validates :address,         presence: true
   belongs_to :user
+  has_many :reservations
   has_many :photos, :dependent => :destroy, inverse_of: :home
   accepts_nested_attributes_for :photos, :allow_destroy => true
+
+  def upcoming_reservations
+    self.reservations.where("checkout > ?", Date.today)
+  end
+
```
Comment (theonlyrao) [inline app/models/home.rb]: this is really clean code!

two thoughts for future reference:
1) we can dry out this and the same in the user mode
2) we need to test the edge cases for reservations on the same day - your solution is way better than what we had in toolchest though

**Comment by theonlyrao** [comment]
Looks great. Also ran the tests and confirmed that we are still passing.
